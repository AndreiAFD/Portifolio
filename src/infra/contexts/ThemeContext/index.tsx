import React, { createContext, useState } from "react";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../core/theme/theme";

enum ThemeEnum {
  light = "light",
  dark = "dark",
}

interface AppThemeContextProps {
  theme?: string;
  handleChangeTheme?: (theme: ThemeEnum) => void;
}

export const AppThemeContext = createContext<AppThemeContextProps>({});

export const AppThemeContextProvider = ({ children }: { children: any }) => {
  const [actualTheme, setActualTheme] = useState<ThemeEnum>(ThemeEnum.light);

  const chakraTheme = extendTheme({
    colors: {
      ...theme[actualTheme],
    },
  });

  const handleChangeTheme = (nextTheme: ThemeEnum) => {
    setActualTheme(nextTheme);
  };

  return (
    <AppThemeContext.Provider
      value={{
        theme: actualTheme,
        handleChangeTheme,
      }}
    >
      <ThemeProvider theme={theme[actualTheme] as typeof theme["dark"]}>
        <ChakraProvider theme={chakraTheme}>{children}</ChakraProvider>
      </ThemeProvider>
    </AppThemeContext.Provider>
  );
};
