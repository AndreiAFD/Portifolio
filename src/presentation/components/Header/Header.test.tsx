import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { forEachChild } from "typescript";
import { Header } from ".";
import { AppThemeContextProvider } from "../../../infra/contexts/ThemeContext";

const ComponentToTest = () => {
  return (
    <BrowserRouter>
      <AppThemeContextProvider>
        <Header />
      </AppThemeContextProvider>
    </BrowserRouter>
  );
};

describe("Header component", () => {
  it("should render correctly", () => {
    render(<ComponentToTest />);
    const header = screen.getByTestId("header-test-id");
    expect(header).toMatchSnapshot();
  });

  it("should click on portifolio link", () => {
    render(<ComponentToTest />);

    const portifolioLink = screen.getByTestId("portifolio-link-testid");
    fireEvent.click(portifolioLink);
  });

  it("should click on portifolio link", () => {
    render(<ComponentToTest />);

    const aboutme = screen.getByTestId("aboutme-link-testid");
    fireEvent.click(aboutme);
  });

  it("should click on contact link", () => {
    render(<ComponentToTest />);

    const contact = screen.getByTestId("contact-link-testid");
    fireEvent.click(contact);
  });
});
