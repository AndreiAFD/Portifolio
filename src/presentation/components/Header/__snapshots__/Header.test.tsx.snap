// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div
  class="sc-bczRLJ jUBbnS"
  data-testid="header-test-id"
>
  <p
    class="chakra-text css-1q74v14"
  >
    <p
      class="chakra-text css-1mjrvz0"
    >
      &lt;
    </p>
    Frontend Developer
    <p
      class="chakra-text css-1g4gjga"
    >
      /
    </p>
    <p
      class="chakra-text css-xdk3wv"
    >
      &gt;
    </p>
  </p>
  <div
    class="sc-gsnTZi jMsOSA"
  >
    <div
      class="sc-dkzDqf cpCdiA active"
    >
      <p
        class="chakra-text css-ahjfv1"
      >
        <a
          data-testid="aboutme-link-testid"
          href="/"
        >
          Sobre
        </a>
      </p>
    </div>
    <div
      class="sc-dkzDqf cpCdiA"
    >
      <p
        class="chakra-text css-ahjfv1"
      >
        <a
          data-testid="portifolio-link-testid"
          href="/portifolio"
        >
          Portifolio
        </a>
      </p>
    </div>
    <div
      class="sc-dkzDqf cpCdiA"
    >
      <p
        class="chakra-text css-ahjfv1"
      >
        <a
          data-testid="contact-link-testid"
          href="/contact"
        >
          Contato
        </a>
      </p>
    </div>
  </div>
</div>
`;
